[DO NOT MERGE] Work-in-progress PWA reference content (#1429)
MicrosoftDocs/edge-developer: microsoft-edge/progressive-web-apps-chromium/index.md

PWAs are websites that are **[progressively enhanced][AListApartUnderstandingProgressiveEnhancement]** to function like native apps on supporting platforms, and like regular web sites on other browsers.

The qualities of a PWA combine **the best of the web and native apps**. PWAs run in browsers, like websites, but have access to native app features like the ability to work offline, be installed as a native app, support push notifications and periodic updates, access hardware features, and more.

**PWAs are first-class citizens on Windows**. When installed, they behave like other apps. They can be added to the Start Menu, pinned to the Taskbar, handle files, run on user login, and more.

PWAs can also be **submitted to the Microsoft Store** where millions of Windows users can discover and easily install them alongside other Windows apps.

:::image type="complex" source="./media/windows-apps-and-store.png" alt-text="PWAs on Windows, in Microsoft Edge, and the Microsoft Store" lightbox="./media/windows-apps-and-store.png":::
    PWAs on Windows, in Microsoft Edge, and the Microsoft Store
:::image-end:::

## Characteristics of a PWA

> PWAs have the same reach as websites: they can be indexed by search engines, can be linked to, and work on all devices from the same codebase. As such they have a much lower development cost than native apps that require a specific codebase  for Android, iOS, and various desktop operating systems.

Reviewer: I’m hearing some pushback on the term "native" from an inclusivity perspective. Alternative could be "proprietary" or "compiled."

Author: Thanks! I wasn't aware. I made changes to all instances of "native" in this page, and used a different phrasing depending on the context.

> :::row:::

Reviewer: Visually, the gap between the subtitle text and the descriptions are far more substantial than the space below the description, almost giving you the impression that the descriptions are associated with the image blow. Can we fix that at all?

Author: Good point. Yes, I can simplify the formatting of this grid and bring things closer together.

ms.technology: pwa
keywords: progressive web apps, PWA, Edge, JavaScript, UWP, Microsoft Store
---
# Overview of Progressive Web Apps (PWAs)

**[Progressive Web Apps][MDNApps]** (PWAs) provide access to open web technologies for cross-platform interoperability and provide your users with an app-like experience customized for their devices.

PWAs are websites that are **[progressively enhanced][AListApartUnderstandingProgressiveEnhancement]** to function like installed apps on supporting platforms, and like regular web sites on other browsers.

The qualities of a PWA combine **the best of the web and compiled apps**. PWAs run in browsers, like websites, but have access to app features like the ability to work offline, be installed on the operating system, support push notifications and periodic updates, access hardware features, and more.

**PWAs are first-class citizens on Windows**. When installed, they behave like other apps. They can be added to the Start Menu, pinned to the Taskbar, handle files, run on user login, and more.

PWAs can also be **submitted to the Microsoft Store** where millions of Windows users can discover and easily install them alongside other Windows apps.

> :::image type="complex" source="./media/windows-apps-and-store.png" alt-text="PWAs on Windows, in Microsoft Edge, and the Microsoft Store" lightbox="./media/windows-apps-and-store.png":::

Reviewer 2: ```suggestion
<!-- draft text, check/revise -->
The following image shows several types of apps that are implemented as a Progressive Web App.  The upper left shows a PWA on Windows.  The upper right shows a PWA in Microsoft Edge.  The bottom image shows the Microsoft Store, implemented as a PWA.
:::image type="complex" source="./media/windows-apps-and-store.png" alt-text="PWAs on Windows, in Microsoft Edge, and the Microsoft Store" lightbox="./media/windows-apps-and-store.png":::
```

Author: I ended up breaking this image into 4 separate images and used a grid to have them displayed side by side.
This way I can use the long description to explain what these images are about.

While most consumers have migrated to mobile-first ways of consuming content, businesses still mostly operate on desktop.
Building your desktop app as a PWA allows you to make use of all the benefits that PWAs come with while targeting those users.

With many new web capabilities and new features in Microsoft Edge and Windows, there are great opportunities to build immersive experiences that deeply integrate with the desktop operating system. Only using web technologies also ensures that your app will work in other browsers, operating systems and devices.

You might not have to use proprietary technologies at all as the web has grown capabilities to deal with file systems, video, audio, high-performing code, databases, USB, Bluetooth, and more.

With a shared codebase between your website, mobile app, and desktop app (across operating systems), you cut down on development cost.

Your app can also be published on the [Microsoft Store][PwaMicrosoftStore], making it more familiar and trustworthy for Windows users to discover and install.

> ## Bridging the gap to desktop apps

Reviewer 2: ```suggestion
## Bridging the gap between web and desktop apps
```

There are many new web capabilities built-in to Microsoft Edge that can make your app feel more integrated on Desktop and offer your users with engaging experiences.

With you PWAs, you can:

*   Handle files.
*   Share content with other apps.
*   Access the clipboard.
*   Sync data and fetch resources in the background.
*   Access device hardware such as Bluetooth and USB.
*   Store content in databases.
*   Take advantage of hardware accelerated graphics.
*   Use CSS layouts, animations, and filters to create advanced designs.
*   Run near-compiled performance code with WebAssembly.

> There isn't much you can't do today with Web technologies, and thanks to Microsoft Edge, PWAs on desktop can take full advantage of this to bridge the gap to what users expect desktop apps to be capable of.

Reviewer 2: "gap to" doesn't work, "gap between"

Navigate to [Myth Busting PWAs][Davrous20191018MythBustingPwasNewEdgeEdition] for more information about what PWAs can do.

> ## The Microsoft Store

Reviewer 2: Need link in this section
[Microsoft Store][PwaMicrosoftStore]

Because PWAs are first-class citizens in the Microsoft Store, users can fully engage with them, from discovery, to installation, to execution, without ever opening the browser.

As the most used app on PCs, the Microsoft Store provides a trustworthy and familiar experience for your users to install your app. Additionally, you can view detailed usage statistics and charts that let you know how your apps in the Microsoft Store are doing.

## Success stories

> Using PWA technologies is a great way to make your app **safe**, **discoverable**, **linkable**, **easy to install** and update, **responsive**, and **network independent**. Many businesses use PWAs and have had success stories.

Reviewer 2: either "have had success" or "have success stories"

        **[Network Independent][MDNPwaAdvantagesNetworkIndependent]**  
        Works offline and in low-network conditions
    :::column-end:::
    :::column:::
        :::image type="icon" source="./media/i_progressive-small.png":::

        **[Progressive][MDNPwaAdvantagesProgressive]**  
        Experience scales up (or down) with device capabilities
    :::column-end:::
    :::column:::
        :::image type="icon" source="./media/i_security-small.png":::

        **[Safe][MDNPwaAdvantagesSafe]**  
        Provides a secure HTTPS endpoint and other user safeguards
    :::column-end:::
:::row-end:::
:::row:::
    :::column:::
        :::image type="icon" source="./media/i_responsive-small.png":::

        **[Responsive][MDNPwaAdvantagesResponsive]**  
        Adapts to the user's screen size or orientation and input method
    :::column-end:::
    :::column:::
        :::image type="icon" source="./media/i_link-small.png":::

        **[Linkable][MDNPwaAdvantagesLinkable]**  
        Share and launch from a standard hyperlink
    :::column-end:::
    :::column:::
    :::column-end:::
:::row-end:::

## PWAs on desktop

> PWAs aren't only a way for websites to appear on mobile home screens. They're based on standard, cross-browser, web technologies that allow them to be installed and to run everywhere.

Reviewer 3: Instead of "aren't only," maybe "are more than"?

Reviewer 4: Agreed and more ... I re-wrote both sentences. Is it an improvement?

Reviewer 3: LGTM

> PWAs aren't only a way for websites to appear on mobile home screens. They're based on standard, cross-browser, web technologies that allow them to be installed and to run everywhere.

Reviewer 4: ```suggestion
PWAs are not limited to displaying websites on mobile home screens. PWAs are based on standard, cross-browser, web technologies that allow them to be installed and to run in many different environments.
```

> While most consumers have migrated to mobile-first ways of consuming content, businesses still mostly operate on desktop. Building your desktop app as a PWA allows you to make use of all the benefits that PWAs come with while targeting those users.

Reviewer 3: I'm not sure about this framing - I might try to word it slightly differently to avoid getting in a desktop-first vs mobile-first war. Maybe something like:

"In recent years, desktop browsing has grown, and desktop computing continues to be the primary productivity environment for many users. Fortunately, PWAs can provide fully tailored, installable experiences on a desktop operating system that inherit the benefits of that environment, while still working across platform from the same code and servers that power your in-browser and mobile experiences."

    :::column-end:::
:::row-end:::

## PWAs on desktop

PWAs aren't only a way for websites to appear on mobile home screens. They're based on standard, cross-browser, web technologies that allow them to be installed and to run everywhere.

While most consumers have migrated to mobile-first ways of consuming content, businesses still mostly operate on desktop.
Building your desktop app as a PWA allows you to make use of all the benefits that PWAs come with while targeting those users.

With many new web capabilities and new features in Microsoft Edge and Windows, there are great opportunities to build immersive experiences that deeply integrate with the desktop operating system and feel native. Only using web technologies also ensures that your app will work in other browsers, operating systems and devices.

You might not have to use native technologies at all as the web has grown capabilities to deal with file systems, video, audio, high-performing code, databases, USB, Bluetooth, and more.

> With a shared codebase between your website, mobile app and desktop app (across operating systems), you cut down on development cost.

Reviewer: What’s our position on the Oxford comma? (I always forget.)

Reviewer 2: https://docs.microsoft.com/en-us/style-guide/top-10-tips-style-voice#remember-the-last-comma
Remember the last comma
In a list of three or more items, include a comma before the conjunction. (The comma that comes before the conjunction is known as the Oxford or serial comma.) To learn more, see Commas.  https://docs.microsoft.com/en-us/style-guide/punctuation/commas

Example
Replace this: Android, iOS and Windows

With this: Android, iOS, and Windows

Your app can also be published on the [Microsoft Store][PwaMicrosoftStore], making it more familiar and trustworthy for Windows users to discover and install.

## Bridging the gap to native

There are many new web capabilities built-in to Microsoft Edge that can make your app look and feel more native on Desktop and offer your users with engaging experiences.

With you PWAs, you can:

> *   Handle files.

Reviewer: suggest semicolons over periods (with the exception of the last item) or no punctuation at all.

Author: I think the Microsoft Writing Style Guide actually advises to use periods in this case. If all items were very short (2 or 3 words each), then the guide suggests to drop punctuation, but this isn't the case here.

*   Share content with other apps.
*   Access the clipboard.
*   Data sync and fetch work in the background.
*   Access device hardware such as Bluetooth and USB.
*   Store content in databases.
*   Take advantage of hardware accelerated graphics.

> *   Use CSS layouts, animations, and filters to achieve a desktop look.

Reviewer: What is a "desktop look"?

Reviewer 2: not sure of context, but consider the term "UX" or "user experience" or "look and feel".  

Finding: the Style Guide doesn't contain "look and feel" and barely mentions "UX".

Author: Made some changes here. Will push a new commit soon.

*   Run near-native performance code with WebAssembly.

There isn't much you can't do today with Web technologies, and thanks to Microsoft Edge, PWAs on desktop can take full advantage of this to bridge the gap to what users expect native apps to be capable of.

Navigate to [Myth Busting PWAs][Davrous20191018MythBustingPwasNewEdgeEdition] for more information about what PWAs can do.

## The Microsoft Store

Because PWAs are first-class citizens in the Microsoft Store, users can fully engage with them, from discovery, to installation, to execution, without ever opening the browser.

> As the most used app on PCs, the Microsoft Store provides a trustworthy and familiar experience for your users to install your app. Additionally, it provides you with analytics.

Reviewer: "Additionally, it provides you with analytics." feels awkward. I don’t know if there is something we can do to add to this sentence that provides more detail around the benefits, but the sentence could use more _umph_.

Reviewer 2: consider:   "usage analytics".    Look at an analytics page at the store and give hints here about what valuable info is there.  Or maybe this set of articles already describes that and you can copy that, summarizing here.

Author: I've made a change which I'll push in a minute. But maybe @jgw96 can take a look at it and suggest better phrasing.

## Success stories

Using PWA technologies is a great way to make your app **safe**, **discoverable**, **linkable**, **easy to install** and update, **responsive**, and **network independent**. Many businesses use PWAs and have had success stories.

*   The Starbucks PWA has increased daily active users twice and orders on desktop are nearly the same rate as mobile ([source][StarbucksSuccessStory]).
*   Trivago saw an increase of 150% for people who add its PWA to the home screen and the increased engagement led to a 97% increase in clickouts to hotel offers ([source][TrivagoSuccessStory]).
*   Tinder cut load times from 11.91 seconds to 4.68 seconds with their PWA and the app is 90% smaller than the native Android app ([source][TinderSuccessStory]).

> To read more success stories, navigate to [PWA Stats][PwaStats].

Reviewer: "navigate to" feels so formal. Could we say something like "check out" or "head over to" instead?

Reviewer 2: agree, will message you offline
